app/js/index.js

```diff
@@ -39,18 +51,42 @@
 		console.log('Allowing browser to handle: ' + JSON.stringify(openRequest));
 	});
 
-	var isMinimized = false;
+	var isHidden = false;
+
+	win.on('hide', function () {
+		isHidden = true;
+	});
 	win.on('minimize', function () {
-		isMinimized = true;
+		isHidden = true;
+	});
+	win.on('show', function () {
+		isHidden = false;
 	});
 	win.on('restore', function () {
-		isMinimized = false;
+		isHidden = false;
 	});
+
+	var windowShowCommand = program.minimizeToTray ? 'show' : 'restore';
+	var windowHideCommand = program.minimizeToTray ? 'hide' : 'minimize';
+
+	function showWindow() {
+		win[windowShowCommand]();
+		isHidden = false;
+	};
+	function hideWindow() {
+		win[windowHideCommand]();
+		isHidden = true;
+	};
+
+  gui.App.on('open', function (cmdline) {
+		showWindow();
+  });
```

Comment:
Ah, I see. This works because we are forcing `single-instance: true`. That feels like a hack on the open event which is intended to handle opening of new files in an application:

https://github.com/nwjs/nw.js/wiki/App#open

Can you elaborate more on when you use this behavior? Are there any other alternatives that we could use?